src/main/protocol-impl/java/com/mysql/cj/protocol/a/authentication/AwsIamAuthenticationTokenHelper.java

```diff
@@ -85,17 +85,25 @@ private Region getRdsRegion() {
     // Check Hostname
     final Pattern auroraDnsPattern =
         Pattern.compile(
-            "(.+)\\.(proxy-|cluster-|cluster-ro-|cluster-custom-)?[a-zA-Z0-9]+\\.([a-zA-Z0-9\\-]+)\\.rds\\.amazonaws\\.com",
+            "(.+)\\.(proxy-|cluster-|cluster-ro-|cluster-custom-)?[a-zA-Z0-9]+\\.([a-zA-Z0-9\\-]+)\\.rds\\.amazonaws\\.com(\\.cn)?",
```

Comment:
Why do we need the last (\\.cn) in this regexp? The code checks for China url pattern further.

Reply:
I'd suggest to replocate entire method getRdsRegion() to RdsUtils.

Reply:
I was double checking this DNS pattern against the one in `RdSUtils.java` and it contains `(\.cn)?` at the end. Same thing in the `aws-advanced-jdbc-wrapper`. So it looks like this pattern may have been missing it.